presto-in-memory/src/test/java/com/facebook/presto/plugin/inmemory/TestInMemorySmoke.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.plugin.inmemory;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.metadata.QualifiedObjectName;
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.testing.MaterializedRow;
+import com.facebook.presto.testing.QueryRunner;
+import com.google.common.collect.Iterables;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.sql.SQLException;
+import java.util.List;
+
+import static com.facebook.presto.plugin.inmemory.InMemoryQueryRunner.createQueryRunner;
+import static java.lang.String.format;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
+
+@Test(singleThreaded = true)
+public class TestInMemorySmoke
+{
+    private QueryRunner queryRunner;
+
+    @BeforeTest
+    public void setUp()
+            throws Exception
+    {
+        queryRunner = createQueryRunner();
+    }
+
+    @AfterTest
+    public void tearDown()
+    {
+        assertThatNoInMemoryTableIsCreated();
+        queryRunner.close();
+    }
+
+    @BeforeMethod
+    public void methodSetUp()
+    {
+        assertThatNoInMemoryTableIsCreated();
+    }
+
+    @Test
+    public void createTableWhenTableIsAlreadyCreated()
+            throws SQLException
+    {
+        String createTableSql = "CREATE TABLE nation as SELECT * FROM tpch.tiny.nation";
+        queryRunner.execute(createTableSql);
+        try {
+            queryRunner.execute(createTableSql);
+            fail("Expected exception to be thrown here!");
+        }
+        catch (RuntimeException ex) { // it has to RuntimeException as FailureInfo$FailureException is private
+            assertTrue(ex.getMessage().equals("line 1:1: Destination table 'inmemory.default.nation' already exists"));
+        }
+        finally {
+            assertThatQueryReturnsValue("DROP TABLE nation", true);
```

Comment:
Ok, but it's outdated since `drop table` is not supported, so there is no `finally` block in more recent version.